drivers/avr/spi_master.c

```diff
@@ -47,57 +47,66 @@ void spi_init(void) {
     SPCR = (_BV(SPE) | _BV(MSTR));
 }
 
-void spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint8_t divisor) {
-    if (currentSlavePin == NO_PIN && slavePin != NO_PIN) {
-        if (lsbFirst) {
-            currentSlaveConfig |= _BV(DORD);
-        }
+bool spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint16_t divisor) {
+    if (currentSlavePin != NO_PIN || slavePin == NO_PIN) {
+        return false;
+    }
 
-        switch (mode) {
-            case 1:
-                currentSlaveConfig |= _BV(CPHA);
-                break;
-            case 2:
-                currentSlaveConfig |= _BV(CPOL);
-                break;
-            case 3:
-                currentSlaveConfig |= (_BV(CPOL) | _BV(CPHA));
-                break;
-        }
+    currentSlaveConfig = _BV(SPE) | _BV(MSTR);
```

Comment:
I don't think we should be doing this here - it's already done in `spi_init()`. That's why I specifically did `SPCR &= ~(currentSlaveConfig)` instead of `SPCR = 0` in `spi_stop()`

Reply:
Rolled back.